odoo/addons/base/res/res_users.py

```diff
@@ -492,6 +498,18 @@ def authenticate(cls, db, login, password, user_agent_env):
                     _logger.exception("Failed to update web.base.url configuration parameter")
         return uid
 
+    @classmethod
+    def check_active(cls, uid):
+        """Verifies that the given uid is authorized to log-in and return a boolean according to the authorization"""
+        cr = cls.pool.cursor()
+        try:
+            self = api.Environment(cr, uid, {})[cls._name]
+            user = self.search([('id', '=', uid)])
+            return bool(user and user.active)
+        finally:
+            cr.close()
+        return False
+
```

Comment:
Maybe put this method after `def check`, to be consistent with the block added in `security.py`, `check_user_active`,  which is after `def check`